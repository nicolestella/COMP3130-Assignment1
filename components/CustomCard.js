// This is the card that will show a travel spot in the travels screen.

import React from "react";
import { Card, Title, Paragraph, Button } from "react-native-paper";

export default function CustomCard({ travelSpot, onPress }) {
	const imgSrc =
		"https://source.unsplash.com/collection/" + travelSpot.collection;

	return (
		<Card style={{ marginBottom: 20, width: "95%" }} onPress={onPress}>
			{/* The image */}
			<Card.Cover source={{ uri: imgSrc }} />
			<Card.Content>
				{/* The title */}
				<Title>{travelSpot.title}</Title>
				{/* The description */}
				<Paragraph numberOfLines={2}>{travelSpot.description}</Paragraph>
			</Card.Content>
			<Card.Actions>
				{/* The save button */}
				<Button mode='contained'>Save</Button>
				<Button>Learn more</Button>
			</Card.Actions>
		</Card>
	);
}
