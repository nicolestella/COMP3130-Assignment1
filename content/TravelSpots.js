module.exports = [
	{
		id: "0",
		title: "Disneyland California",
		keyword: "theme park",
		collection: "3446667",
		tags: ["USA", "Theme Park", "Family"],
		description:
			"Disneyland, amusement park in Anaheim, California, featuring characters, " +
			"rides, and shows based on the creations of Walt Disney and the Disney Company. " +
			"Though its central building, the Sleeping Beauty Castle, is modeled on Germany’s " +
			"Neuschwanstein Castle, it is an unmistakable icon of American popular culture. " +
			"Disneyland is the only Disney theme park designed and built under the supervision " +
			"of Walt Disney.",
	},
	{
		id: "1",
		title: "Sydney Opera House",
		keyword: "art",
		collection: "9740328",
		tags: ["Australia", "Sightseeing", "Art"],
		description:
			"The Sydney Opera House is a multi-venue performing arts centre at Sydney Harbour " +
			"located in Sydney, New South Wales, Australia. It is one of the 20th century's " +
			"most famous and distinctive buildings. Designed by Danish architect Jørn Utzon, " +
			"but completed by an Australian architectural team headed up by Peter Hall, " +
			"the building was formally opened on 20 October 1973 after a gestation beginning " +
			"with Utzon's 1957 selection as winner of an international design competition.",
	},
	{
		id: "2",
		title: "Marina Bay Sands",
		keyword: "hotel",
		collection: "6803095",
		tags: ["Singapore", "Hotel", "Family"],
		description:
			"Marina Bay Sands stands out as a stunning icon of Singapore’s ever-growing city skyline. " +
			"The 3 skyscrapers have – what seems to be – a cruise ship balancing precariously across the top. " +
			"In fact, this is one of the highlights of Marina Bay Sands Hotel: a huge rooftop pool with sweeping " +
			"views of Singapore, with an observation deck and a fancy rooftop club at one end. Besides a stunning " +
			"5-star hotel with over 2,500 rooms, other highlights include a luxury shopping mall with faux-canals " +
			"and Italian gondola rides, Singapore’s largest ice-skating rink, 2 nightclubs, a host of fantastic restaurants, " +
			"and one of the world’s most expensive casinos.",
	},
	{
		id: "3",
		title: "Very Cool Restaurant",
		keyword: "food",
		collection: "4308781",
		tags: ["Worldwide", "Food", "Drinks"],
		description:
			"Experience the best of Michelin starred dining Worldwide, with exquisite dishes highlighting modern " +
			"French-inspired cuisine, delivered with flawless service in beautiful surroundings. Our flagship destination, Very Cool " +
			"Restaurant in Chelsea, gained the ultimate accolade of three Michelin stars in 2001 and has retained them ever since, a " +
			"true mark of excellence, quality and consistency. Named after one of the world’s finest wines, Michelin starred Pétrus by " +
			"Gordon Ramsay is the perfect choice for lunch or dinner while enjoying the exceptional wine list which features world class vintages " +
			"including those of Château Pétrus.",
	},
];
