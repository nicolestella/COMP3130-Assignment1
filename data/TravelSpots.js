module.exports = [
	{
		id: "0",
		title: "Disneyland California",
		country: "USA",
		keyword: "theme park",
		collection: "3446667",
		tags: ["Fun", "Theme Park", "Family"],
		description:
			"Disneyland, amusement park in Anaheim, California, featuring characters, " +
			"rides, and shows based on the creations of Walt Disney and the Disney Company. " +
			"Though its central building, the Sleeping Beauty Castle, is modeled on Germany’s " +
			"Neuschwanstein Castle, it is an unmistakable icon of American popular culture. " +
			"Disneyland is the only Disney theme park designed and built under the supervision " +
			"of Walt Disney.",
	},
	{
		id: "1",
		title: "Sydney Opera House",
		country: "Australia",
		keyword: "art",
		collection: "9740328",
		tags: ["Urban", "Sightseeing", "Art"],
		description:
			"The Sydney Opera House is a multi-venue performing arts centre at Sydney Harbour " +
			"located in Sydney, New South Wales, Australia. It is one of the 20th century's " +
			"most famous and distinctive buildings. Designed by Danish architect Jørn Utzon, " +
			"but completed by an Australian architectural team headed up by Peter Hall, " +
			"the building was formally opened on 20 October 1973 after a gestation beginning " +
			"with Utzon's 1957 selection as winner of an international design competition.",
	},
	{
		id: "2",
		title: "Marina Bay Sands",
		country: "Singapore",
		keyword: "hotel",
		collection: "6803095",
		tags: ["Food", "Hotel", "Urban"],
		description:
			"Marina Bay Sands stands out as a stunning icon of Singapore’s ever-growing city skyline. " +
			"The 3 skyscrapers have – what seems to be – a cruise ship balancing precariously across the top. " +
			"In fact, this is one of the highlights of Marina Bay Sands Hotel: a huge rooftop pool with sweeping " +
			"views of Singapore, with an observation deck and a fancy rooftop club at one end. Besides a stunning " +
			"5-star hotel with over 2,500 rooms, other highlights include a luxury shopping mall with faux-canals " +
			"and Italian gondola rides, Singapore’s largest ice-skating rink, 2 nightclubs, a host of fantastic restaurants, " +
			"and one of the world’s most expensive casinos.",
	},
	{
		id: "3",
		title: "Very Cool Restaurant",
		country: "Worldwide",
		keyword: "food",
		collection: "4308781",
		tags: ["Fun", "Food", "Drinks"],
		description:
			"Experience the best of Michelin starred dining Worldwide, with exquisite dishes highlighting modern " +
			"French-inspired cuisine, delivered with flawless service in beautiful surroundings. Our flagship destination, Very Cool " +
			"Restaurant in Chelsea, gained the ultimate accolade of three Michelin stars in 2001 and has retained them ever since, a " +
			"true mark of excellence, quality and consistency. Named after one of the world’s finest wines, Michelin starred Pétrus by " +
			"Gordon Ramsay is the perfect choice for lunch or dinner while enjoying the exceptional wine list which features world class vintages " +
			"including those of Château Pétrus.",
	},
	{
		id: "4",
		title: "Grand Canyon",
		country: "USA",
		keyword: "nature",
		collection: "4252461",
		tags: ["Family", "Nature", "Sightseeing"],
		description:
			"The Grand Canyon is a mile-deep gorge in northern Arizona. Scientists estimate the canyon may have formed 5 to 6 million years ago " +
			"when the Colorado River began to cut a channel through layers of rock. Humans have inhabited the area in and around the canyon since " +
			"the last Ice Age. The first Europeans to reach the Grand Canyon were Spanish explorers in the 1540s. President Benjamin Harrison first " +
			"protected the Grand Canyon in 1893 as a forest reserve, and it became an official United States National Park in 1919. Nearly two billion " +
			"years of Earth's geological history have been exposed as the Colorado River and its tributaries cut their channels through layer after " +
			"layer of rock while the Colorado Plateau was uplifted.",
	},
	{
		id: "5",
		title: "Great Barrier Reef",
		country: "Australia",
		keyword: "nature",
		collection: "3345483",
		tags: ["Family", "Nature", "Sightseeing"],
		description:
			"One of Australia’s most remarkable natural gifts, the Great Barrier Reef is blessed with the breathtaking beauty of the world’s largest coral " +
			"reef. The reef contains an abundance of marine life and comprises of over 3000 individual reef systems and coral cays and literally hundreds of " +
			"picturesque tropical islands with some of the worlds most beautiful sun-soaked, golden beaches. The Great Barrier Reef is one of the seven wonders " +
			"of the natural world, and pulling away from it, and viewing it from a greater distance, you can understand why. It is larger than the Great Wall " +
			"of China and the only living thing on earth visible from space.",
	},
	{
		id: "6",
		title: "Singaporean Ramen",
		country: "Singapore",
		keyword: "food",
		collection: "57856056",
		tags: ["Fun", "Food", "Drinks"],
		description:
			"Presenting a new wave of modern Japanese cuisine, Akira Back thrills diners with the best of classic Japanese dishes woven with " +
			"ingredients and flavours from Chef Akira Back's South Korean heritage as well as culinary influences and experiences from his upbringing in USA. " +
			"Expect quality ingredients, cross-cultural culinary techniques, and a party in the mouth at Akira Back. One of 16 restaurants in major cities " +
			"worldwide including Las Vegas, Toronto, Seoul, Dubai, Hanoi, Bangkok and Singapore.",
	},
	{
		id: "7",
		title: "A Hotel Somewhere",
		keyword: "hotel",
		country: "Worldwide",
		collection: "46255628",
		tags: ["Food", "Hotel", "Drinks"],
		description:
			"A Hotel Somewhere offers a front row seat to the corners of the globe. Discover your destination’s iconic landmarks, traditional celebrations and " +
			"cultural delights - all with the peace-of-mind in knowing that A Hotel Somewhere is always at your fingertips. Stay well rested for your adventure with " +
			"comfortable Grand Beds band spa-inspired bathrooms, all while staying connected to your destination. Enjoy luxurious in-room amenities and indulge in " +
			"all that our iconic destinations have to offer to make your getaway Grand. For the discerning traveler, the A Hotel Somewhere offers a great place for " +
			"respade and relaxation. To provide an even more seamless experience, this private social space furnishes express check-ins, daily continental " +
			"breakfasts and specialty craft cocktail hours. When booking your reservation, ask about upgrading to a Grand Club room.",
	},
];
