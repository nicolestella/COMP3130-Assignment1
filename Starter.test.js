import React from 'react';
import renderer from 'react-test-renderer';
import CustomCard from './components/CustomCard';
import TravelSpots from './data/TravelSpots';

test("CustomCard will render", () => {
  const json = renderer.create(
    <CustomCard travelSpot={TravelSpots[0]} />
  ).toJSON();
  expect(json.props.style.marginTop).toBe(20);
  expect(json.props.style.width).toBe('95%');
});

test("CustomCard will contain custom information", () => {
  const customSpot = {
    id: "0",
		title: "Random title",
		country: "Somewhere",
		keyword: "Word",
		collection: "3446667",
		tags: ["A", "B", "C"],
    description: "Some description",
  }
  const json = renderer.create(
    <CustomCard travelSpot={customSpot} />
  ).toJSON();
  expect(json.children.includes("Random title"));
  expect(json.children.includes("Somewhere"));
  expect(json.children.includes("Some description"));
});

// Snapshot tests
test("CustomCard renders correctly when it is not editable", () => {
  const json = renderer.create(
    <CustomCard travelSpot={TravelSpots[0]} />
  ).toJSON();
  expect(json).toMatchSnapshot();
});

test("CustomCard renders correctly when it is editable", () => {
  const customSpot = {
    id: "0",
		title: "Random title",
		country: "Somewhere",
		keyword: "Word",
		collection: "3446667",
		tags: ["A", "B", "C"],
    description: "Some description",
    custom: true
  }
  const json = renderer.create(
    <CustomCard travelSpot={customSpot} />
  ).toJSON();
  expect(json).toMatchSnapshot();
});

