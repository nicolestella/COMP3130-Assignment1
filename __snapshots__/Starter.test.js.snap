// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCard renders correctly when it is editable 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 4,
      "elevation": 1,
      "marginTop": 20,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0.5,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 0.75,
      "width": "95%",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#eeeeee",
            "height": 195,
            "overflow": "hidden",
          },
          Object {
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          undefined,
        ]
      }
    >
      <Image
        siblings={
          Array [
            "withTheme(Card.Cover)",
            "withTheme(Card.Title)",
            "Card.Content",
            "Card.Actions",
          ]
        }
        source={
          Object {
            "uri": "https://source.unsplash.com/collection/3446667",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "height": undefined,
              "justifyContent": "flex-end",
              "padding": 16,
              "resizeMode": "cover",
              "width": undefined,
            },
            Object {
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
          },
          Object {
            "minHeight": 72,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  Array [
                    Object {
                      "minHeight": 30,
                    },
                    Object {
                      "marginBottom": 0,
                    },
                    undefined,
                  ],
                ],
              ],
            ]
          }
        >
          Random title
        </Text>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 12,
                    "letterSpacing": 0.4,
                    "lineHeight": 20,
                    "marginVertical": 2,
                  },
                  Array [
                    Object {
                      "marginVertical": 0,
                      "minHeight": 20,
                    },
                    undefined,
                  ],
                ],
              ],
            ]
          }
        >
          Somewhere
        </Text>
      </View>
      <View />
    </View>
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
          },
          undefined,
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                undefined,
              ],
            ],
          ]
        }
      >
        Some description
      </Text>
    </View>
    <View
      index={3}
      siblings={
        Array [
          "withTheme(Card.Cover)",
          "withTheme(Card.Title)",
          "Card.Content",
          "Card.Actions",
        ]
      }
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 8,
          },
          undefined,
        ]
      }
      total={4}
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "minWidth": "auto",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Learn more
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#6200ee",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 2,
            "minWidth": "auto",
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0.75,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#ffffff",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Edit
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`CustomCard renders correctly when it is not editable 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 4,
      "elevation": 1,
      "marginTop": 20,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0.5,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 0.75,
      "width": "95%",
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#eeeeee",
            "height": 195,
            "overflow": "hidden",
          },
          Object {
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
          },
          undefined,
        ]
      }
    >
      <Image
        siblings={
          Array [
            "withTheme(Card.Cover)",
            "withTheme(Card.Title)",
            "Card.Content",
            "Card.Actions",
          ]
        }
        source={
          Object {
            "uri": "https://source.unsplash.com/collection/3446667",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "height": undefined,
              "justifyContent": "flex-end",
              "padding": 16,
              "resizeMode": "cover",
              "width": undefined,
            },
            Object {
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 16,
          },
          Object {
            "minHeight": 72,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "500",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 20,
                    "letterSpacing": 0.15,
                    "lineHeight": 30,
                    "marginVertical": 2,
                  },
                  Array [
                    Object {
                      "minHeight": 30,
                    },
                    Object {
                      "marginBottom": 0,
                    },
                    undefined,
                  ],
                ],
              ],
            ]
          }
        >
          Disneyland California
        </Text>
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.54)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 12,
                    "letterSpacing": 0.4,
                    "lineHeight": 20,
                    "marginVertical": 2,
                  },
                  Array [
                    Object {
                      "marginVertical": 0,
                      "minHeight": 20,
                    },
                    undefined,
                  ],
                ],
              ],
            ]
          }
        >
          USA
        </Text>
      </View>
      <View />
    </View>
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 16,
          },
          undefined,
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontFamily": "System",
                "fontWeight": "400",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontSize": 14,
                  "letterSpacing": 0.25,
                  "lineHeight": 20,
                  "marginVertical": 2,
                },
                undefined,
              ],
            ],
          ]
        }
      >
        Disneyland, amusement park in Anaheim, California, featuring characters, rides, and shows based on the creations of Walt Disney and the Disney Company. Though its central building, the Sleeping Beauty Castle, is modeled on Germany’s Neuschwanstein Castle, it is an unmistakable icon of American popular culture. Disneyland is the only Disney theme park designed and built under the supervision of Walt Disney.
      </Text>
    </View>
    <View
      index={3}
      siblings={
        Array [
          "withTheme(Card.Cover)",
          "withTheme(Card.Title)",
          "Card.Content",
          "Card.Actions",
        ]
      }
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 8,
          },
          undefined,
        ]
      }
      total={4}
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "minWidth": "auto",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    Object {
                      "marginHorizontal": 8,
                    },
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Learn more
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
